### PR TITLE
Content assist participants and messages

### PR DESCRIPTION
**Pull request by MalinaStefaniaStoicanescu**
Add content assist for all participants (to propose when create a
message) and content assist & validation for all messages types

Signed-off-by: MalinaStefaniaStoicanescu <malina.stoicanescu@thalesgroup.com>

### CODE REVIEW

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/src/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.xtend [hunk 1]
```diff
@@ -50,56 +56,107 @@ class DslProposalProvider extends AbstractDslProposalProvider {
...
+
+	override completeSequenceMessage_Name(EObject model, Assignment assignment, ContentAssistContext context,
+		ICompletionProposalAcceptor acceptor) {
+		for (String el : messagesDefinedBefore(model as SequenceMessage)) {
+			acceptor.accept(createCompletionProposal("\"" + el + "\"", "\"" + el + "\"", null, context))
+
 		}
 	}
 
+	def createAcceptorForParticipants(EObject el, ICompletionProposalAcceptor acceptor, ContentAssistContext context) {
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/src/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.xtend:123]: Here can cast el as Participant and do only (el as Participant).name

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java [hunk 2]
```diff
@@ -95,18 +101,110 @@ public void getExistingParticipants(final String keyword, final ContentAssistCon
...
   
+  @Override
+  public void completeSequenceMessage_Name(final EObject model, final Assignment assignment, final ContentAssistContext context, final ICompletionProposalAcceptor acceptor) {
+    List<String> _messagesDefinedBefore = this.messagesDefinedBefore(((SequenceMessage) model));
+    for (final String el : _messagesDefinedBefore) {
+      acceptor.accept(this.createCompletionProposal((("\"" + el) + "\""), (("\"" + el) + "\""), null, context));
+    }
+  }
+  
+  public void createAcceptorForParticipants(final EObject el, final ICompletionProposalAcceptor acceptor, final ContentAssistContext context) {
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java:124]: Simplyfy this function, no need for all the ifs.
Can use the Participant class.

File: plugins/org.polarsys.capella.scenario.editor/src/org/polarsys/capella/scenario/editor/helper/EmbeddedEditorInstanceHelper.java [hunk 3]
```diff
@@ -59,18 +62,37 @@
...
     List<CapellaElement> exchanges = SelectInvokedOperationModelForSharedDataAndEvent
         .getAvailableExchangeItems(sourceIR, targetIR, false);
-
-    for (CapellaElement message : exchanges) {
-      if (message instanceof ExchangeItemAllocation) {
-        ExchangeItemAllocation allocation = (ExchangeItemAllocation) message;
-        if (allocation.getAllocatedItem() instanceof ExchangeItem) {
-          messagesName.add(allocation.getAllocatedItem().getName());
+    if (!exchanges.isEmpty()) {
+      for (CapellaElement message : exchanges) {
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor/src/org/polarsys/capella/scenario/editor/helper/EmbeddedEditorInstanceHelper.java:85]: There is a need for this code or the code on the else branch could work?

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/src/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.xtend (hunk 1 above)
Comment (MalinaStefaniaStoicanescu) [reply plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/src/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.xtend]: done

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java (hunk 2 above)
Comment (MalinaStefaniaStoicanescu) [reply plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java]: done

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java [hunk 4]
```diff
@@ -95,18 +94,46 @@ public void getExistingParticipants(final String keyword, final ContentAssistCon
...
-      acceptor.accept(this.createCompletionProposal(((Actor) el).getName(), ((Actor) el).getName(), null, context));
+      String _name = ((Participant) el).getName();
+      String _plus = ("\"" + _name);
+      String _plus_1 = (_plus + "\"");
+      String _name_1 = ((Participant) el).getName();
+      String _plus_2 = ("\"" + _name_1);
+      String _plus_3 = (_plus_2 + "\"");
+      acceptor.accept(
+        this.createCompletionProposal(_plus_1, _plus_3, 
+          null, context));
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java:121]: can also update this version with _plus_x variables and do as for participants instead.

File: plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java [hunk 5]
```diff
@@ -95,18 +94,46 @@ public void getExistingParticipants(final String keyword, final ContentAssistCon
   public void completeSequenceMessage_Source(final EObject model, final Assignment assignment, final ContentAssistContext context, final ICompletionProposalAcceptor acceptor) {
     EList<Participant> _variablesDefinedBefore2 = this.variablesDefinedBefore2(((Model) model));
     for (final EObject el : _variablesDefinedBefore2) {
-      acceptor.accept(this.createCompletionProposal(((Actor) el).getName(), ((Actor) el).getName(), null, context));
+      String _name = ((Participant) el).getName();
+      String _plus = ("\"" + _name);
+      String _plus_1 = (_plus + "\"");
+      String _name_1 = ((Participant) el).getName();
+      String _plus_2 = ("\"" + _name_1);
+      String _plus_3 = (_plus_2 + "\"");
```
Comment (georgiana-ecobici) [inline plugins/org.polarsys.capella.scenario.editor.dsl/org.polarsys.capella.scenario.editor.dslscenario.ui/xtend-gen/org/polarsys/capella/scenario/editor/dslscenario/ui/contentassist/DslProposalProvider.java:102]: can also update this version with _plus_x variables and do as for participants instead.

**Review by georgiana-ecobici — APPROVED** [review]

**Pull request merged**